internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -306,9 +306,15 @@ func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
 
 					wpid, err := syscall.Wait4(-1, &status, syscall.WNOHANG, nil)
 					if wpid <= 0 || err != nil {
+						// We break the loop since an error occurred
 						break
 					}
-					if wpid == cmd.Process.Pid && err == nil {
+
+					if wpid <= 0 {
```

Comment:
this condition will be never met, you can remove it

Reply:
Ooops, I was clearly not paying enough attention, thanks for catching this. Fixed.